pkg/cmd/controller/csr.go

```diff
@@ -10,37 +11,66 @@ import (
 )
 
 const (
-	controllerName                    = "csr-approver-controller"
-	monitoringServiceAccountNamespace = "openshift-monitoring"
-	monitoringServiceAccountName      = "cluster-monitoring-operator"
-	monitoringCertificateSubject      = "CN=system:serviceaccount:openshift-monitoring:prometheus-k8s"
-	monitoringLabelKey                = "metrics.openshift.io/csr.subject"
-	monitoringLabelValue              = "prometheus"
+	controllerNamePrefix                  = "csr-approver-controller"
+	monitoringNamespace                   = "openshift-monitoring"
+	monitoringRequesterServiceAccountName = "cluster-monitoring-operator"
+	monitoringSubjectNameLabelKey         = "metrics.openshift.io/csr.subject"
 )
 
+type monitoringCSR struct {
+	subject     string
+	certSubject string
+	selector    labels.Selector
+}
+
+func newMonitoringCSR(subjectName, subjectServiceAcount string) (*monitoringCSR, error) {
+	labelsRequirement, err := labels.NewRequirement(monitoringSubjectNameLabelKey, selection.Equals, []string{subjectName})
+	if err != nil {
+		return nil, err
+	}
+	return &monitoringCSR{
+		subject:     subjectName,
+		selector:    labels.NewSelector().Add(*labelsRequirement),
+		certSubject: fmt.Sprintf("CN=system:serviceaccount:%s:%s", monitoringNamespace, subjectServiceAcount),
+	}, nil
+}
+
+// RunCSRApproverController approves the CSRs for prometheus and metrics-server client certs that are used for authn against various /metrics endpoints.
```

Comment:
should I mention that this runs two controllers now?

Reply:
Worth adding, yes